sql/api/src/main/scala/org/apache/spark/sql/MergeIntoWriter.scala

```diff
@@ -202,7 +202,7 @@ abstract class MergeIntoWriter[T] {
  * @tparam T
  *   The type of data in the MergeIntoWriter.
  */
-case class WhenMatched[T] private[sql] (
+case class WhenMatched[T](
```

Comment:
Such a change has led to an enlargement of the access scope of the constructor, which may not conform to the original intended expectations. Isn't there a better alternative solution?

Reply:
Reverted